### PR TITLE
Installing node from images.json

### PR DESCRIPTION
**Pull request by pacostas**
## Summary

This PR selects which node-engine to install based on the node versions that exist on `images.json` file.
 
## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: generate_test.go [hunk 1]
```diff
@@ -268,265 +104,197 @@ func testGenerate(t *testing.T, context spec.G, it spec.S) {
...
 		it.After(func() {
 			Expect(os.RemoveAll(workingDir)).To(Succeed())
+			Expect(os.RemoveAll(imagesJsonTmpDir)).To(Succeed())
 		})
 
 		it("Specific version of node requested", func() {
 
-			extensionToml, _ := readExtensionTomlTemplateFile()
+			imagesJsonContent := utils.GenerateImagesJsonFile([]string{"16", "18"}, []bool{false, true}, false)
+			hello := fmt.Sprintf(imagesJsonContent)
```
Comment (pacostas) [inline generate_test.go:113]: remove

File: internal/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	}
+
+	configTomlContent, err := CreateConfigTomlFileContent(defaultNodeVersion, nodejsStacks, stackId)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	configTomlContentString := configTomlContent.Bytes()
+	return configTomlContentString, nil
+}
```
Comment (pacostas) [inline internal/utils/utils.go:63]: Added a wrapper, and for that wrapper on the utils tests we only test that it manages to return the result and that it manages to propagate an error. The type of the error, the type of the result and the edge cases are covered from the tests of the functions that are wrapped.

File: integration/fetch_run_image_from_env.go [hunk 3]
```diff
@@ -52,57 +39,29 @@ func testFetchRunImageFromEnv(t *testing.T, context spec.G, it spec.S) {
...
 					settings.Buildpacks.NodeEngine.Online,
 					settings.Buildpacks.Processes.Online,
 				).
-				WithEnv(map[string]string{"BP_UBI_RUN_IMAGE_OVERRIDE": nodeRunImage, "BP_NODE_VERSION": fmt.Sprint(buildNodeMajorVersion)}).
+				WithEnv(map[string]string{"BP_UBI_RUN_IMAGE_OVERRIDE": nodeRunImage}).
 				WithPullPolicy("always").
 				Execute(name, source)
 
-			Expect(err).NotTo(HaveOccurred(), logs.String())
+			Expect(err).To(HaveOccurred())
```
Comment (pacostas) [inline integration/fetch_run_image_from_env.go:57]: The function errors after the extension has run due to it tries to download the fake image `nodeRunImage := "this-is-a-run-image"`. Therefore the build fails, so I changed it to only check the logs and that the build failed instead of checking the logs and checking and that the build that succeeded. If we can find an image that will always be available, we can have also the build to succeed. In any case the extension is tested properly so to me seems like there is no issue of test coverage on this scenario.

**Comment by pacostas** [comment]
@mhdawson The tests do not pass due to there is no new ubi-base image with the images.json file in it. If you force the release on the ubi-base-stack, it will generate one, which will update the builder and the tests of this PR will pass.

File: internal/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+	"strings"
+	"text/template"
+
+	"github.com/paketo-community/ubi-nodejs-extension/structs"
+
+	"github.com/BurntSushi/toml"
+)
+
+var DEFAULT_USER_ID = 1002
+var DEFAULT_GROUP_ID = 1000
```
Comment (mhdawson) [inline internal/utils/utils.go:22]: I wonder if these could be in a common place as we define them in multiple files?

**Comment by mhdawson** [comment]
Just one tiny question, otherwise looks good.

Pushed through a release of the stack, will re-run the tests.

**Comment by pacostas** [comment]
Yes, you are right. On the generate.go these constants are not being used anymore as the `getDuringBuildPermissions` function has been moved to utils package. Although I moved these constants to a new package called `constants` as are a bit hidden under the utils package. I can easily revert these changes by moving these constants under the utils package file if you think is not on the context of this PR.
Also I added this commit https://github.com/paketo-community/ubi-nodejs-extension/pull/203/commits/15c94001180126fd86b4c72f3becf9cf88a01a28 which removes the tmp directory by using the builtin function for making tmp directories which also removes them automatically.

**Review by mhdawson — APPROVED** [review]
LGTM
